raiden/network/transport/matrix.py

```diff
@@ -697,7 +695,7 @@ def _receive_delivered(self, delivered: Delivered):
             return
 
         self._raiden_service.handle_state_change(
-            ReceiveDelivered(delivered.delivered_message_identifier),
+            ReceiveDelivered(delivered.sender, delivered.delivered_message_identifier),
```

Comment:
Maybe it'd be useful to add this handling on `message_handler.py::on_message`? Even though it's also possibly used internally by the transport (e.g. udp using it to close the queue/retry), seems like this part of the logic can be handled similarly for both transports. The returned `bool` can then be used to indicate if there was the corresponding message in some queue, triggering the internal cleaning.

Reply:
I am not sure i completely follow you're idea... but i understand that you want `handle_message_*` to return booleans to `on_message` which is returned also to the transport. This returned bool can be used to indicate to the transport whether to cleanup it's internal state or not?

Reply:
No, to put it in `handle_message_*` would need to duplicate the checks. The idea would be to put it in the beginning of `on_message`, as it's shared between the transports, and it already returns a bool, so it's just a matter of `return False` early if it's one of these validated messages without the corresponding queued message, and on transport, just skip the cleanup if the return was not truty.

Reply:
Yes, sure, then these loops should be taken out of the transports, but `UDPTransport` currently depends on the `Delivered` to set some async_result. It should instead happen when the message is cleared of the respective queue, like is done in Matrix:
https://github.com/raiden-network/raiden/blob/f762dbd69e9f77532816b1dd78b7056e9dc9a5d2/raiden/network/transport/matrix.py#L769-L784

Reply:
I checked out how what you described can be implemented.
First, let me try to make you visualize the execution flow:
```
Transport -> on_message -> handle_message_* -> RaidenService::handle_state_change -> WAL -> node::state_transition -> node::(handle_processed | handle_delivered)
```
This is the call flow. Which means that if i wanted to make on_message return `True` based on whether the processed / delivered messages were processed correctly by the node.py handler, i would have to update all methods `handle_mesage_*` in `message_handler.py` to return `True/False` based on the result. That being said, `RaidenService::handle_state_change` returns the resulting events list from the state change which means that i will either have to introduce an event type just to indicate the result of `Processed` or `Delivered` processing for the `handle_message_*` to return `True/False`. 
I hope the above makes sense to you, otherwise, please follow the call flow and think about what i am seeing. TBH, I am not sure this change would be worth the effort.